open_optical_gating/cli/cli.py

```diff
@@ -336,6 +369,9 @@ def pog_state(self, frame):
             self.frame_history = np.roll(self.frame_history, -1, axis=0)
 
         # Gets the argmin of SAD and adds to frame_history array
+        # TODO: JT writes: I don't know what this history is used for, but I think it's pretty weird to have a preallocated buffer,
+        # and then just keep updating the final element if we overrun the buffer. I would have expected a FIFO buffer containing up to N recent frames...
+        # That would also simplify subsequent logic e.g. "Predicting with partial buffer" would not be needed

```

Comment:
I believe I copied this verbatim from your ObjC codes? (Caveat: three years ago)

Reply:
Ah, my concern about "just keep updating the final element" was because I'd missed the earlier np.roll! At least I agree it is correct now. Still find the need for "predicting with partial buffer" a bit unsatisfactory, but this is just a cosmetic preference of mine.